proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,342 @@
+
+A validation solution follows from an invalid parenting event to occur, generally a `component` or `assembly` being nested below a non-`group`. The user must be prompted to decide whether to change the `kind` of the nested asset (to something like `subcomponent`) or change the `kind` of any non-`group` ancestors.
+
+However, users and tools generally expect to be able to parent `Xform`s underneath other `Xform`s without too much trouble (ie. an extra validation step). They also generally expect that such changes are local and invertible. Reparenting a scope and then undoing it (manually not through an undo queue) will leave spurious `group` tags on any ancestors or an unintentionally "demoted" model to `subcomponent`. Validating on the parenting event can lead to both cruft and blocking user decisions.
+
+Prior to path expressions, delaying validation until `Save`-ing of a stage was perhaps viable. Not necessarily ideal that model hierarchy would be a broken state, but the impact would be limited to a session's interactive features and not imaging. Delaying validation as a post-process could lead to different material binding and light collections. A user saving a stage they are happy with and then validation informing them of errors that change the way the stage renders risks erroding user trust of validation. It also could lead to users avoiding building expressions around model hierarchy features. As model hierarchy is intended to be efficiently cached, relying on adhoc tags or naming conventions could reduce overall performance of path expressions.
+
+Self assembling model hierarchy does not remove the need for validation of model hierarchy. However, it focuses it on the scopes that are changing and reduces the number of cases that can be considered invalid. We enumerate the potential invalid cases below.
+
+### Invalid Case #1: Missing root prim `kind`
+As a concession to performance and "pay for what you use", a hierarchy must opt into model hierarchy at the root prim. Validating and repairing a missing root prim kind has the same risks issues and without automatic group propagation. We suggest that users, tools, and pipelines are willing to treat root prims as special and more willing to carefully maintain and validate their correctness than intermediate scopes.
+### Invalid Case #2: Nested `component` models
+When `component` (or `assembly`) models are nested, the general recommendation is to turn the nested model into a `subcomponent`. Importantly, following this recommendation will not impact path expression matching. Model hierarchy population should have already discarded nested models in both. Converting them to `subcomponent` shores up existing model hierarchy correctness, but does not change the "model"-ness of any ancestors or descendants. (_NOTE-- It's possible that path expressions may match against subcomponents, and so a more conservative validation recommendation may be to convert their kind to the empty string._)
```

Comment:
There is no such recommendation for nested `assembly` models - assemblies of assemblies are expected and not uncommon in our pipeline.

Regarding the recommendation about fixing component-under-component to the empty string... despite the fact that it could change pattern evaluation, `component` carries some amount of importance with it, and demoting it to "nothing" seems undesirable... possibly this is one of those validation fixer cases where a choice of fix is needed?